src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -429,38 +429,43 @@ private static ValidationProblem CreateValidationProblem(IdentityResult result)
         // We expect a single error code and description in the normal case.
         // This could be golfed with GroupBy and ToDictionary, but perf! :P
         Debug.Assert(!result.Succeeded);
-        var errorDictionary = new Dictionary<string, string[]>(1);
+        var errorDictionary = AggregateDictionary(result.Errors.Select(error => (error.Code, error.Description)));
+        return TypedResults.ValidationProblem(errorDictionary);
+    }
+
+    private static async Task<InfoResponse> CreateInfoResponseAsync<TUser>(TUser user, ClaimsPrincipal claimsPrincipal, UserManager<TUser> userManager)
+        where TUser : class
+    {
+        return new()
+        {
+            Email = await userManager.GetEmailAsync(user) ?? throw new NotSupportedException("Users must have an email."),
+            IsEmailConfirmed = await userManager.IsEmailConfirmedAsync(user),
+            Claims = AggregateDictionary(claimsPrincipal.Claims.Select(claim => (claim.Type, claim.Value))),
```

Comment:
Why not return the `IEnumerable<(TKey, TValue)>` directly? Creating the dictionary seems like unnecessary work.

Reply:
This is similar to my alternative proposal in my API review update comment at https://github.com/dotnet/aspnetcore/issues/49424#issuecomment-1736506896. The main difference is that I defined a new `ClaimInfo` type rather than use a `ValueTuple<string, string>` because S.T.J does not serialize ValueTuples well. You can use `IncludeFields = true` for it to kinda work, but it will still use `Item1` and `Item2` for the property names even if you explicitly specify something else.

But even with the `ClaimInfo`, I think a flat list of claims is harder to consume than a dictionary of arrays.

For example, the `IEnumareable<ClaimInfo>` produces the following JSON for two roles:

```json
[
  {
    "Type": "role",
    "Value": "role1"
  },
  {
    "Type": "role",
    "Value": "role2"
  }
]
```

vs. the dictionary:

```
{
  "role": [
    "role1",
    "role2"
  ]
}
```

And the JS code to check the claims to see if a user is in a role is a lot simpler if it's a dictionary rather than a flat list:

```js
claims.some(c => c.Type == "role" && c.Value == "role1")
```

vs. the dictionary

```js
claims.role.includes("role1")
```